File: nodejs/bubble.js
function bubble(n) {
    while(true){
        pronto = true;
        for (let i = 1; i < n.length; i++) { 
            if (n[i - 1] > n[i]) {
                pronto = false;
                t = n[i];
                n[i] = n[i - 1];
                n[i - 1] = t;
            }
        }
        if (pronto) break;
    }
    return n;
}

File: ruby/bubble.rb
def bubble(n)
  loop do
    pronto = true
      (n.size).times do |i| 
        if i != 0
          if n[i - 1] > n[1]
            pronto = false
            temp = n[i]
            n[i] = n[i - 1]
            n[i - 1] = temp
          end
        end
      end
    break if true
  end 
  n
end
p bubble([1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11])
 